### DISCUSSION TITLE
Multiple sim testing

### DISCUSSION THREAD

**Discussion by roeiedri** [Chooloo/koler#341] [Q&A] [answered]
Hey hey
I need some help testing Koler with multiple sim devices.
Since it's not really possible to truly emulate with more than 1 sim, a physical device is needed. 

Ill upload here new apks to test with fixes for reports.

**Comment by jeevanpulluru** [comment]
I'm in...

**Comment by mojienjoyment** [comment]
im ready for that

**Comment by roeiedri** [comment]
[koler-release.zip](https://github.com/Chooloo/koler/files/8147061/koler-release.zip)

Pls test this new apk :)
Using multiple sim phone:
- Make a call (without having any default sim account in phone settings)
- Make a call (having a default sim account in settings and seeing it uses it without asking)
- Get a call 

I added a phone account selection dialog when making a call

@Mojienjoynent @jeevanpulluru

**Reply by jeevanpulluru** [reply]
Hi, it's not working. Running on pixel 6 (Android 12L Beta 3). I do get the dialog but there's no option to select the sim. Although I've already set the default option to ask me to select the sim whenever I place a cal...
![Screenshot_20220226-224521~2.png](https://user-images.githubusercontent.com/53634434/155852604-e87610f6-a2c3-47f9-a1ad-d700a1f723ed.png)
![Screenshot_20220226-224608.png](https://user-images.githubusercontent.com/53634434/155852620-39d86225-13b0-40fb-bd0d-1b51fa349105.png)

**Reply by jeevanpulluru** [reply]
Before placing the call, I do get the dialog to select the sim, but again as you see there's another dialog to select the sim and it's the same story again...

**Reply by jeevanpulluru** [reply]
https://user-images.githubusercontent.com/53634434/155853427-187fda67-7cf3-4832-8528-1d90eb3c9ee8.mp4
    Although selecting a default sim in settings and placing a call does work...

**Reply by roeiedri** [reply]
Ok so the dialog works but is empty (doesnt suggest any accounts).
Ill look into that.
@jeevanpulluru

**Reply by roeiedri** [reply]
> https://user-images.githubusercontent.com/53634434/155853427-187fda67-7cf3-4832-8528-1d90eb3c9ee8.mp4
>     Although selecting a default sim in settings and placing a call does work...

Ohhh ok so the problem is the sim dialog, i need to remove it.

**Reply by jeevanpulluru** [reply]
Yes, please look into the video I've uploaded, the first dialog is more than sufficient for selecting the sim but in call UI there seems to be another dialog which is not required...

**Reply by roeiedri** [reply]
[koler-release.zip](https://github.com/Chooloo/koler/files/8149199/koler-release.zip)

Please try this, I removed sim dialog and calling with bundle.
@jeevanpulluru

**Reply by jeevanpulluru** [reply]
https://user-images.githubusercontent.com/53634434/155895205-f2c8df57-6ebe-4eeb-a8c9-d3541b4c5d18.mp4

Not working...

**Reply by jeevanpulluru** [reply]
Select sim dialogue is not appearing...

**Reply by mojienjoyment** [reply]
> [koler-release.zip](https://github.com/Chooloo/koler/files/8147061/koler-release.zip)
> 
> Pls test this new apk :) Using multiple sim phone:
> 
> * Make a call (without having any default sim account in phone settings)
> * Make a call (having a default sim account in settings and seeing it uses it without asking)
> * Get a call
> 
> I added a phone account selection dialog when making a call
> 
> @Mojienjoynent @jeevanpulluru

Device : poco x3 pro miui 12.5 android 11
I test it for this first release:
Make a call (without having any default sim account in phone settings):

Opens dialog box for account selection but there isn't any account for selection like this screenshot:
![Screenshot_2022-02-28-11-20-26-579_com chooloo www koler](https://user-images.githubusercontent.com/68327458/155945298-685d2d37-0e98-4b9a-8a07-c8dc34a1e3fa.jpg)

But for:
Make a call (having a default sim account in settings and seeing it uses it without asking):
When I select default sim, it calls perfectly.. Without any problem with that Sim...

**Reply by mojienjoyment** [reply]
> [koler-release.zip](https://github.com/Chooloo/koler/files/8149199/koler-release.zip)
> 
> Please try this, I removed sim dialog and calling with bundle. @jeevanpulluru


Same situation... 
Same problem in this release too....
Without having default sim in phone settings - > empty dialog box appears
With having default sim - > MAKING CALL perfectly

**Reply by roeiedri** [reply]
> > [koler-release.zip](https://github.com/Chooloo/koler/files/8149199/koler-release.zip)
> > 
> > Please try this, I removed sim dialog and calling with bundle. @jeevanpulluru
> 
> 
> Same situation... 
> Same problem in this release too....
> Without having default sim in phone settings - > empty dialog box appears
> With having default sim - > MAKING CALL perfectly

Did the default sim work also before the apk i sent? Like in the official release

**Reply by jeevanpulluru** [reply]
> > > [koler-release.zip](https://github.com/Chooloo/koler/files/8149199/koler-release.zip)
> > > 
> > > Please try this, I removed sim dialog and calling with bundle. @jeevanpulluru
> > 
> > 
> > Same situation... 
> > Same problem in this release too....
> > Without having default sim in phone settings - > empty dialog box appears
> > With having default sim - > MAKING CALL perfectly
> 
> Did the default sim work also before the apk i sent? Like in the official release

Yes..if we select the default sim from system settings it works flawlessly...

**Reply by mojienjoyment** [reply]
> > > [koler-release.zip](https://github.com/Chooloo/koler/files/8149199/koler-release.zip)
> > > Please try this, I removed sim dialog and calling with bundle. @jeevanpulluru
> > 
> > 
> > Same situation...
> > Same problem in this release too....
> > Without having default sim in phone settings - > empty dialog box appears
> > With having default sim - > MAKING CALL perfectly
> 
> Did the default sim work also before the apk i sent? Like in the official release

Yes...

**Reply by roeiedri** [reply]
That's so weird, because even Google's dialer uses the same method as I do to get the phone accounts,
Is any of you an android developer by any change?
@Mojienjoynent @jeevanpulluru

**Reply by roeiedri** [reply]
[koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
 One last chance with the original method

**Reply by jeevanpulluru** [reply]
> That's so weird, because even Google's dialer uses the same method as I do to get the phone accounts,
> Is any of you an android developer by any change?
> @Mojienjoynent @jeevanpulluru 

I'm not but you can try to contact simple apps developer if you want...

**Reply by jeevanpulluru** [reply]
> [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
>  One last chance with the original method
> 

Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.

**Reply by roeiedri** [reply]
> > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> >  One last chance with the original method
> > 
> 
> Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.

Can I ask for a quick screen video of that situation?

**Reply by jeevanpulluru** [reply]
> > > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> > >  One last chance with the original method
> > > 
> > 
> > Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.
> 
> Can I ask for a quick screen video of that situation?

https://photos.app.goo.gl/LwyX9CX7c65r1Xr68
Follow the link for the video...

**Reply by roeiedri** [reply]
> > > > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> > > >  One last chance with the original method
> > > > 
> > > 
> > > Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.
> > 
> > Can I ask for a quick screen video of that situation?
> 
> https://photos.app.goo.gl/LwyX9CX7c65r1Xr68
> Follow the link for the video...

No video in the link :)

**Reply by jeevanpulluru** [reply]
Please add an option so that we can see from which sim we've got the call...

**Reply by roeiedri** [reply]
> > > > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> > > >  One last chance with the original method
> > > > 
> > > 
> > > Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.
> > 
> > Can I ask for a quick screen video of that situation?
> 
> https://photos.app.goo.gl/LwyX9CX7c65r1Xr68
> Follow the link for the video...

Waiting for the video :)

**Reply by jeevanpulluru** [reply]
> > > > > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> > > > >  One last chance with the original method
> > > > > 
> > > > 
> > > > Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.
> > > 
> > > Can I ask for a quick screen video of that situation?
> > 
> > https://photos.app.goo.gl/LwyX9CX7c65r1Xr68
> > Follow the link for the video...
> 
> Waiting for the video :)

Please click ont the link for the recorded video. I'm unable to send it here due to size restrictions...

**Reply by roeiedri** [reply]
> > > > > > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> > > > > >  One last chance with the original method
> > > > > > 
> > > > > 
> > > > > Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.
> > > > 
> > > > Can I ask for a quick screen video of that situation?
> > > 
> > > https://photos.app.goo.gl/LwyX9CX7c65r1Xr68
> > > Follow the link for the video...
> > 
> > Waiting for the video :)
> 
> Please click ont the link for the recorded video. I'm unable to send it here due to size restrictions...

It says the album is empty

**Reply by jeevanpulluru** [reply]
> > > > > > > [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip)
> > > > > > >  One last chance with the original method
> > > > > > > 
> > > > > > 
> > > > > > Success it's working although there's one tiny bug. While selecting the sim, I have to select the sim twice before it can place a call. Since, I had to select the sim selection twice before I place the call I've selected sim 1 first and then if I select sim 2nd sim second time it selects only sim 1 for making the call.
> > > > > 
> > > > > Can I ask for a quick screen video of that situation?
> > > > 
> > > > https://photos.app.goo.gl/LwyX9CX7c65r1Xr68
> > > > Follow the link for the video...
> > > 
> > > Waiting for the video :)
> > 
> > Please click ont the link for the recorded video. I'm unable to send it here due to size restrictions...
> 
> It says the album is empty

Kindly check once...

**Reply by mojienjoyment** [reply]
> [koler-release.zip](https://github.com/Chooloo/koler/files/8155441/koler-release.zip) One last chance with the original method

This version Okey but with a little bug that jeevanpulluru mentioned!

**Comment by roeiedri** [comment]
[koler-release.zip](https://github.com/Chooloo/koler/files/8173027/koler-release.zip)

Another shot, I think I found the issue, I hope so
@jeevanpulluru

**Reply by jeevanpulluru** [reply]
> [koler-release.zip](https://github.com/Chooloo/koler/files/8173027/koler-release.zip)
> 
> Another shot, I think I found the issue, I hope so
> @jeevanpulluru 

Works like a charm. Everything's perfect. Now all that needs are some additional features and some marketing. 
Happy coding@roeiedri 😁😁😁

**Reply by jeevanpulluru** [reply]
> > [koler-release.zip](https://github.com/Chooloo/koler/files/8173027/koler-release.zip)
> > 
> > Another shot, I think I found the issue, I hope so
> > @jeevanpulluru 
> 
> Works like a charm. Everything's perfect. Now all that needs are some additional features and some marketing. 
> Happy coding@roeiedri 😁😁😁

https://user-images.githubusercontent.com/53634434/156488686-9bdb14a0-48fa-4c4e-a3e1-d76a03200526.mp4

**Reply by roeiedri** [reply]
Yyyayyyyyyyyyyyy 
Thank you for your help!

**Reply by mojienjoyment** [reply]
> [koler-release.zip](https://github.com/Chooloo/koler/files/8173027/koler-release.zip)
> 
> Another shot, I think I found the issue, I hope so @jeevanpulluru

Yeah, This latest version works perfectly! Great!
But one thing..  You can remove "should ask for sim" option in settings.  Because it's useless and haven't any effect on this behavior... We should enable "ask for Sim" option in device settings.
![IMG_20220303_180402](https://user-images.githubusercontent.com/68327458/156585647-0356bc5f-0679-40a4-a2cd-40aa41d74b8c.jpg)

**Reply by jeevanpulluru** [reply]
> Yyyayyyyyyyyyyyy 
> Thank you for your help
sorry, but I found another bug, the orientation is not fixed,  i.e. when I receive a call and the phone is in my pocket, the call UI seems to rotate automatically even though orientation is locked in system settings.

**Reply by roeiedri** [reply]
0d6143434fbf210a3722d7b315e464e361a038c7

Removed "ask sim" preference and fixed sim selection

**Comment by roeiedri** [comment, marked as answer]
0d6143434fbf210a3722d7b315e464e361a038c7

Fixed issue
Thanks to @jeevanpulluru and @Mojienjoynent
